platform/default/src/mbgl/storage/pmtiles_file_source_stub.cpp

```diff
@@ -0,0 +1,38 @@
+#include <mbgl/storage/file_source_request.hpp>
+#include <mbgl/storage/pmtiles_file_source.hpp>
+#include <mbgl/storage/resource.hpp>
+#include <mbgl/util/async_request.hpp>
+
+namespace mbgl {
+
+class PMTilesFileSource::Impl {
```

Comment:
Nice, I think it is better.

Maybe this can be a stub for any file source? Since it is not PMtiles specific.

Reply:
Good idea, maybe in another PR? Or would I do this in this PR?